File: src/main/java/com/github/sorusclient/client/setting/SettingContainer.java
package com.github.sorusclient.client.setting;

import java.util.Map;

public interface SettingContainer {
    String getId();
    void load(Map<String, Object> settings);
    void loadForced(Map<String, Object> settings);
    void removeForced();
    Map<String, Object> save();

    boolean isShared();
    void setShared(boolean isShared);
}

File: src/main/kotlin/com/github/sorusclient/client/setting/SettingContainer.kt
package com.github.sorusclient.client.setting

interface SettingContainer {
    val id: String
    fun load(settings: Map<String, Any>)
    fun loadForced(settings: Map<String, Any>)
    fun removeForced()
    fun save(): Map<String, Any>
    var shared: Boolean
}